pkg/kubelet/network/plugins.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/golang/glog"
+)
+
+const DefaultPluginName = "no_op"
```

Comment:
This just popped into my head - plugin names should be qualified-name format, which means no underscores.  "no-op" is proper.  BUT...

plugin names should also be namespaceable (as in IsQualifiedName) so they don't step on each other.  So this should be "kubernetes.io/no-op".  That begs the question of what the exec plugins should be called?  How do we ensure that a vendor can drop in a network plugin and ensure that they get the name they want?

volume plugins do this by escaping the '/' into '~', so if a user wanted "kubernetes.io/ipvlan" they would say --network_plugin="kubernetes.io/ipvlan" - this implies that you would have to decouple the Name() interface from the file name, or else expect the filename to be an escaped qualified name.

Option 1) When you read the dir, you find dirs like "kubernetes.io~ipvlan".  When you report Name() you unescape that into "kubernetes.io/ipvlan".

Option 2) When you read the dir, you find dirs like "ipvlan".  You have to call `ipvlan name` to get a string that you report as Name()

I prefer option 1 with extra docs in the great big comment at the top of exec.  Your thoughts?